### PR TITLE
Add some simple test cases

### PR DESCRIPTION
**Pull request by sangyx**
I designed a few simple test cases based on the new derivation mechanism we discussed.

### CODE REVIEW

**Comment by hameerabbasi** [comment]
Can we write these as proper `xfail` tests in PyTest?

**Comment by sangyx** [comment]
OK, I will modify it.

File: src/tests/test_builtin_diffs.py [hunk 1]
```diff
@@ -171,3 +171,38 @@ def test_arbitrary_function(backend, func, y_d, domain):
...
+        (onp.arange(12).reshape(2, 3, 2), None, 2, lambda x: np.sum(x, axis=1), [onp.tile([[1, 0], [0, 1]], (2, 3)).reshape(2, 1, 2, 3, 2)], None),
+        (onp.arange(2).reshape(1, 2, 1), onp.arange(2).reshape(1, 1, 2), 2, lambda x, y: np.matmul(x, y), onp.array([[0, 0, 1, 0], [0, 0, 0, 1]]).reshape(1, 2, 2, 2, 1), onp.array([[0, 0, 0, 0], [1, 0, 0, 1]]).reshape(1, 2, 2, 1, 2)),
+        (onp.arange(12).reshape((2, 3, 2)), None, 2, lambda x: x, onp.stack([onp.eye(6).reshape(3, 2, 3, 2), onp.eye(6).reshape(3, 2, 3, 2)]), None)
+    ],
+)
+def test_separation(backend, u, v, diff_ndim, func, diff_u, diff_v):
+    try:
+        with ua.set_backend(backend), ua.set_backend(udiff, coerce=True):
+            u = np.asarray(u)
+            u.var = udiff.Variable('u')
```
Comment (hameerabbasi) [inline src/tests/test_builtin_diffs.py]:
```suggestion
            u.var = udiff.Variable('u', diff_ndim=diff_ndim)
```

File: src/tests/test_builtin_diffs.py [hunk 2]
```diff
@@ -171,3 +171,38 @@ def test_arbitrary_function(backend, func, y_d, domain):
...
+            u.diff_ndim = diff_ndim
+            if v is not None:
+                v = np.asarray(v)
+                v.var = udiff.Variable('v')
```
Comment (hameerabbasi) [inline src/tests/test_builtin_diffs.py]:
```suggestion
                v.var = udiff.Variable('v', diff_ndim=diff_ndim)
```

**Review by hameerabbasi — COMMENTED** [review]
I'd suggest making the `diff_ndim` different for each input too. Perhaps split this into two tests, `test_separation_unary` and `test_separation_binary`?

**Comment by hameerabbasi** [comment]
Thanks, @sangyx!

**Pull request merged**